service/matching/matchingEngine.go

```diff
@@ -211,6 +216,24 @@ func (e *matchingEngineImpl) getTaskListManager(taskList *taskListID,
 	return mgr, nil
 }
 
+func (e *matchingEngineImpl) cacheTaskListByDomain(taskList *taskListID) error {
+	domainName, err := e.domainCache.GetDomainName(taskList.domainID)
+	if err != nil {
+		return err
+	}
+	cachedTL := e.taskListCache.Get(domainName)
+	var taskLists map[string]bool
+	if cachedTL != nil {
+		taskLists = cachedTL.(map[string]bool)
+	} else {
+		taskLists = map[string]bool{}
+	}
+
+	taskLists[taskList.name] = true
```

Comment:
When will cached tasklist be removed? I think matching engine has an idle check for each tasklist and we can remove tasklist from the cache if it's idle for too long and be unloaded.

Reply:
Good point. I think we can support TTL in cache?

Reply:
let's update the cache in removeTaskListManager as well